source/base/parameter_handler.cc

```diff
@@ -1976,6 +2022,19 @@ ParameterHandler::scan_line(std::string        line,
             entries->get<std::string>(path + path_separator + "alias"));
         }
 
+      // Check if the entry (or the resolved alias) is deprecated, print a
+      // warning if it is.
+      if (entries->get_optional<std::string>(path + path_separator +
+                                             "deprecation_status") &&
+          entries->get<std::string>(path + path_separator +
+                                    "deprecation_status") == "true")
+        {
+          AssertThrow(false,
```

Comment:
if you throw here, you never complete the rest of the parsing further down or am I wrong?

Reply:
good catch, the test didnt check if the parameter was read correctly. I modified the flow and the test to make sure the parameter is correctly read.